source/elements/oneMKL/source/domains/dft/dft.rst

```diff
@@ -5,23 +5,181 @@
 
+.. math::
+    z^{m}_{k_1, k_2,\ldots, k_d} = \sigma \displaystyle\sum_{j_d=0}^{n_d-1}\dots\displaystyle\sum_{j_2=0}^{n_2-1}\displaystyle\sum_{j_1=0}^{n_1-1} w^{m}_{j_1, j_2,\dots,j_d} \exp \left[ \delta 2\pi \imath \left( \sum_{\ell=1}^{d} \frac{j_{\ell}k_{\ell}}{n_{\ell}} \right) \right] \ \forall \left(k_1, \ldots, k_d\right) \in \mathbb{Z}^{d}
+    :label: eq_dft_definition
+
+where :math:`\imath^2 = -1` and :math:`\sigma` is a scale factor.
+In :eq:`eq_dft_definition`, :math:`\delta` determines one of the two
+"directions" of the DFT: :math:`\delta=-1` defines the "forward DFT" while
+:math:`\delta=+1` defines the "backward DFT".
+
+The domain of input (resp. output) discrete sequences for a forward (resp.
+backward) DFT is referred to as "forward domain". Conversely, the domain of
+output (resp. input) discrete sequences for forward (resp. backward) DFT is
+referred to as "backward domain".
+
+oneMKL supports single-precision (fp32) and double-precision (fp64)
+floating-point arithmetic for the calculation of DFTs, using two types of
+forward domains:
+
+- the set of complex :math:`d`-dimensional periodic sequences, referred to as
+  "complex forward domain";
+- the set of real :math:`d`-dimensional periodic sequences, referred to as
+  "real forward domain".
+
+Similarly, we refer to DFTs of complex (resp. real) forward domain as "complex
+DFTs" (resp. "real DFTs"). Regardless of the type of forward domain, the
+backward domain's data sequences are alway complex.
+
+The calculation of the same DFT for several, *i.e.*, :math:`M > 1`, data sets of
+the same type of forward domain, using the same precision is referred to as a
+"batched DFT".
+
+.. _onemkl_dft_finite_range_of_indices:
+
+Finite range of indices
++++++++++++++++++++++++
+
+In general, given the periodicity of the discrete data considered in any DFT,
+ranges of indices :math:`\left(k_1, \ldots, k_d\right) \in \mathbb{Z}^{d}` such
+that :math:`0\leq k_{\ell} < n_{\ell}, \forall \ell \in \lbrace 1, \ldots, d
+\rbrace` suffice to determine any relevant :math:`d`-dimensional sequence
+unambiguously (for any valid :math:`m`). In case of real DFTs, the data
+sequences in backward domain can be fully determined from a smaller range of
+indices. Indeed, if all entries of :math:`w` are real in
+:eq:`eq_dft_definition`, then the entries of :math:`z` are complex and, for any
+valid :math:`m`,
+:math:`\left(z^{m}_{k_1, k_2, \dots, k_d}\right)^{*} = z^{m}_{n_1 - k_1, n_2 - k_2, \dots, n_d - k_d}`
+:math:`\forall \left(k_1, k_2, \ldots, k_d\right) \in \mathbb{Z}^{d}` where
+:math:`\lambda^{*}` represents the conjugate of complex number :math:`\lambda`.
+This conjugate symmetry relation makes roughly half the data redundant in
+backward domain: in case of real DFTs, the data sequences in backward domain can
+be fully determined even if one of the :math:`d` indices :math:`k_{\ell}` is
+limited to the range
+:math:`0\leq k_{\ell} \leq \lfloor \frac{n_{\ell}}{2}\rfloor`. In oneMKL, the
```

Comment:
I think you're talking about using $0 \leq k_{\ell} < \lfloor \frac{n_{\ell}}{2}\rfloor + 1$ instead of $0 \leq k_{\ell} \leq \lfloor \frac{n_{\ell}}{2}\rfloor$, am I right? Please correct me if that's not what you meant. For a 1D odd size (*e.g.* $n_{\ell}  =7$), we do indeed need the "mid-point" of index $\lfloor \frac{n_{\ell}}{2}\rfloor$ with indices starting from $0$ (*e.g.* $\lfloor \frac{n_{\ell}}{2}\rfloor =3$) and this range captures it.

In either case, the range contains $\left(\lfloor \frac{n_{\ell}}{2}\rfloor + 1 \right)$ elements (regardless of $n_{\ell}$ being odd or even). I chose to express the range as $0 \leq k_{\ell} \leq \lfloor \frac{n_{\ell}}{2}\rfloor$ everywhere, for personal convenience.

Reply:
Lol thank you, yes, indexing from zero.